collect_app/src/main/java/org/odk/collect/android/utilities/gdrive/GoogleAccountsManager.java

```diff
@@ -170,7 +178,42 @@ public boolean checkAccountPermission() {
     }
 
     public String getSelectedAccount() {
-        return (String) preferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT);
+        Account[] googleAccounts = credential.getAllAccounts();
+        String account = (String) preferences.get(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT);
+
+        if (account == null) {
+            return null;
+        }
+
+        if (googleAccounts != null && googleAccounts.length > 0) {
+            for (Account googleAccount : googleAccounts) {
+                if (googleAccount.name.equals(account)) {
+                    return account;
+                }
+            }
+
+            preferences.save(PreferenceKeys.KEY_SELECTED_GOOGLE_ACCOUNT, null);
```

Comment:
Instead of saving `null`, we can reset this preference using the `reset` method in `GeneralSharedPreferences`. That way it will never be equal to null

Reply:
Oh awesome! Never knew about that method.